patroni/ha.py

```diff
@@ -877,11 +905,21 @@ def _is_healthiest_node(self, members: Collection[Member], check_replication_lag
         members = [m for m in members if m.name != self.state_handler.name and not m.nofailover and m.api_url]
 
         if members:
-            for st in self.fetch_nodes_statuses(members):
+            for st, priority in self.fetch_nodes_statuses_and_priorities(members):
                 if st.failover_limitation() is None:
                     if st.in_recovery is False:
                         logger.warning('Primary (%s) is still alive', st.member.name)
                         return False
+                    if self.is_lagging(st.wal_position):
+                        # The other node appears to have intolerable lag, move on (regardless of priority)
+                        continue
+                    if self.get_effective_tags().get('failover_priority', 1) < priority:
```

Comment:
The `get_effective_tags()` is only useful for `touch_member()`, because sometimes we need to dynamically set `nosync` tag. In all other cases it is better to directly access property in the `Patroni` class.

Reply:
Also, this check must be performed after the next condition: https://github.com/zalando/patroni/pull/2780/files#diff-b6a3a0ef64783a6d32feb908d74a3a009e9f25008b8ca5313c9c9747e3b74a63R923